plugins/native-script-example/src/main/java/org/elasticsearch/examples/nativescript/script/PopularityScoreScriptFactory.java

```diff
@@ -0,0 +1,103 @@
+
+import org.apache.lucene.search.Scorer;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
+import org.elasticsearch.index.fielddata.ScriptDocValues;
+import org.elasticsearch.script.AbstractDoubleSearchScript;
+import org.elasticsearch.script.ExecutableScript;
+import org.elasticsearch.script.NativeScriptFactory;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Factory for the script that boosts score of a record based on a value of  the record's field.
+ * <p>
+ * This native script demonstrates how to write native custom scores scripts.
+ */
+public class PopularityScoreScriptFactory implements NativeScriptFactory {
+
+    @Override
+    public ExecutableScript newScript(@Nullable Map<String, Object> params) {
+        String fieldName = params == null ? null : XContentMapValues.nodeStringValue(params.get("field"), null);
+        if (fieldName == null) {
+            throw new IllegalArgumentException("Missing the field parameter");
+        }
+        return new PopularityScoreScript(fieldName);
+    }
+
+    /**
+     * Indicates if document scores may be needed by the produced scripts.
+     *
+     * @return {@code true} if scores are needed.
+     */
+    @Override
+    public boolean needsScores() {
+        return true;
+    }
+
+    @Override
+    public String getName() {
+        return "popularity";
+    }
+
+    /**
+     * This script takes a numeric value from the field specified in the parameter field. And calculates boost
+     * for the record using the following formula: 1 + log10(field_value + 1). So, records with value 0 in the field
+     * get no boost. Records with value 9 gets boost of 2.0, records with value 99, gets boost of 3, 999 - 4 and so on.
+     */
+    private static class PopularityScoreScript extends AbstractDoubleSearchScript {
+
+        private final String field;
+
+        private Scorer scorer;
+
+        public PopularityScoreScript(String field) {
+            this.field = field;
+        }
+
+        @Override
+        public void setScorer(Scorer scorer) {
+            this.scorer = scorer;
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public double runAsDouble() {
+            try {
+                ScriptDocValues<Long> docValue = (ScriptDocValues<Long>) doc().get(field);
+                if (docValue != null && !docValue.isEmpty()) {
+                    ScriptDocValues.Longs fieldData = (ScriptDocValues.Longs) docValue;
+                    double boost = 1 + Math.log10(fieldData.getValue() + 1);
+                    // Because this script is used in custom_score script the value of score() is populated.
+                    // In all other cases doc().getScore() should be used instead.
+                    return boost * scorer.score();
+
+                }
+                return scorer.score();
+            } catch (IOException ex) {
```

Comment:
I don't think we should give leniency like this as an example. If there is a problem, it should be bubbled up.